dashboard/generate_dashboard_data.py

```diff
@@ -0,0 +1,251 @@
+
+from airtable.airtable import get_all_records
+
+S3_BUCKET = "dxe-backup"
+S3_BACKUP_DIR = "dashboard"
+S3_ACCESS_KEY = os.environ["AIRTABLE_BACKUP_AWS_ACCESS_KEY_ID"]
+S3_SECRET_KEY = os.environ["AIRTABLE_BACKUP_AWS_SECRET_ACCESS_KEY"]
+
+
+def chunks(l, n):
+    """Yield successive n-sized chunks from l."""
+    for i in xrange(0, len(l), n):
+        yield l[i:i+n]
+
+
+def links_to_ids(links):
+    """Uses the Facebook Graph API to get group/page's ids from their links"""
+    # This uses's the batch request feature
+    # (https://developers.facebook.com/docs/graph-api/making-multiple-requests)
+    # to make many logical requests within one http request, saving the overhead
+    # cost
+
+    ids = []
+    for links_subset in chunks(links, 50):  # batch limit is 50
+        reqs = ['{{"method":"GET", "relative_url":"{}"}}'.format(link) for link in links_subset]
+        batch_req = '[' + ",".join(reqs) + ']'
+
+        r = requests.post(
+            "https://graph.facebook.com/v2.5",
+            data={"batch": batch_req},
+            headers={"Authorization": "Bearer {}".format(os.environ["FACEBOOK_APP_ACCESS_TOKEN"])}
+        )
+        data = r.json()
+        for i, res in enumerate(data):
+            community = json.loads(res['body'])['id']
+            if community.startswith("http"):
+                # If the id here is actually a link, then that means it was a fb group. If it's an
+                # id, then it was a fb page. The response format is different for each.
+                community = json.loads(res['body'])['og_object']['id']
+            ids.append(community)
+    return ids
+
+
+def get_facebook_chapter_links():
+    """Grab the facebook links for each local chapter from airtable."""
+    raw_data = get_all_records("Chapters", "Main View")
+    chapter_links = []
+    for row in raw_data:
+        fields = row["fields"]
+        if "Facebook" in fields:
+            chapter_links.append(fields["Facebook"])
+    return list(set(chapter_links))  # some differnent chapters use the same facebook page, remove dups
+
+
+def get_events(_id):
+    """Given the id of a group or page, returns a list of all the events it hosted.
+    Events in the form:
+        {id: <event id>, start_time: <ISO datetime string>, name: <name>}"""
+    events = []
+    next_page_url = "https://graph.facebook.com/v2.5/{}/events?fields=id,start_time,name,owner{{id}}".format(_id)
+    while next_page_url:
+        r = requests.get(
+            next_page_url,
+            headers={"Authorization": "Bearer {}".format(os.environ["FACEBOOK_APP_ACCESS_TOKEN"])}
+        )
+        data = r.json()
+        if 'data' in data and len(data['data']) > 0:
+
+            # some groups share events they don't host, which get returned here
+            events += [d for d in data['data'] if d['owner']['id'] == _id]
+
+            next_page_url = data['paging'].get('next')
+        else:
+            next_page_url = False
+    for e in events:
```

Comment:
```
{
  id: 132343234234,
  start_time: blahhTZ,
  name: "eventname",
  owner: {id: 12341234123}
}
```

so leaving owner in makes each event not a row dict. I only added it (`?fields= ... ,owner{id}`) so I could do the check to make sure the event is owned by the community.